src/utils/EventBus.ts

```diff
@@ -0,0 +1,34 @@
+export type Listener<T extends unknown[] = any[]> = (...args: T) => void;
+
+export default class EventBus<E extends string = string,
+                              M extends { [K in E]: unknown[] } = Record<E, any[]>> {
```

Comment:
```suggestion
export default class EventBus<
    E extends string = string,
    M extends {[K in E]: unknown[]} = Record<E, any[]>
> {
```

Тут кажется форматирование немного поехало;

Reply:
fix